Resolve double encoding problem

vishrutshah opened the pull request:
This is regarding the Issue #360 where users were uploading file with spaces in name then uploaded file has encodided characters e.g. %20 for space. This is due to the double encoding problem (once in [blob_service.rb:1442](https://github.com/Azure/azure-sdk-for-ruby/blob/master/service_management/azure/lib/azure/blob/blob_service.rb#L1442) and second time in [service.rb:40](https://github.com/Azure/azure-sdk-for-ruby/blob/master/service_management/azure/lib/azure/core/service.rb#L40)). 

This change removed encoding from [blob_service.rb:1442](https://github.com/Azure/azure-sdk-for-ruby/blob/master/service_management/azure/lib/azure/blob/blob_service.rb#L1442) and updated other CGI.escapse method to URI.encode method.

azurecla commented:
Hi __@vishrutshah__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

veronicagg commented on service_management/azure/lib/azure/service_bus/auth/shared_access_signer.rb:
```diff
@@ -23,9 +23,9 @@ def name
 
         def token(uri)
-          url_encoded_resource = CGI.escape(uri.to_s.downcase).gsub('+', '%20').downcase
+          url_encoded_resource = URI.escape(uri.to_s.downcase).downcase
```
I was reading about [the difference between uri escape and cgi escape](http://stackoverflow.com/questions/2824126/whats-the-difference-between-uri-escape-and-cgi-escape), have you looked into this? Maybe we should not be using URI.escape?

vishrutshah commented:
@veronicagg Thanks for referencing "[What's the difference between URI.escape and CGI.escape?](http://stackoverflow.com/questions/2824126/whats-the-difference-between-uri-escape-and-cgi-escape)". I've added integration test for the file name with spaces and removed the second encoding. 

@creepycheese Could you please review the PR and make sure that non uri encoded path works for you after removing double encoding? 

Thanks,

vishrutshah commented:
Respective code changes in azure-core gem is at [PR #3](https://github.com/Azure/azure-ruby-asm-core/pull/3).

@veronicagg Could you please review and let me know your feedback.

veronicagg commented on service_management/azure/test/integration/blob/create_page_blob_test.rb:
```diff
@@ -80,4 +80,28 @@
+      blob.name.must_equal blob_name
+    end
+
+    it 'should create block blob with complex in name' do
```
is this validating something additional to "should create a block blob with spaces in name" and "creates page blob with non uri encoded path"?

vishrutshah replied:
The second test validates the blob name with non uri encoded characters like `"with фбаф.txt"` to see that the those get persisted in the blob name on azure and can be passed on through the service.

veronicagg replied:
oh, I see, thanks!

veronicagg commented:
@vishrutshah Looks good! thanks.

vishrutshah commented:
Thanks for the review!

yaxia commented on service_management/azure/test/integration/blob/create_page_blob_test.rb:
```diff
@@ -28,8 +28,8 @@
-      blob = subject.create_page_blob container_name, blob_name, length
+    it 'creates page blob with non uri encoded path' do
+      blob = subject.create_page_blob container_name, 'фбаф.jpg', length
       blob.name.must_equal blob_name
```
Shouldn't it be: `blob.name.must_equal 'фбаф.jpg'` ?

vishrutshah replied:
@yaxia Yest it must be. Currently these tests aren't recorded and running in the travis-ci but i verified that this is wrong.
